.github/workflows/KubernetesExternalClients.yaml

```diff
@@ -43,40 +43,66 @@ jobs:
-        export_default_credentials: true 
+        credentials_json: '${{ secrets.GKE_SA_KEY }}'
+
+    - name: 'Set up Cloud SDK'
+      uses: 'google-github-actions/setup-gcloud@v1'
 
     - name: Install Kubectl
       run: |-
         gcloud components install kubectl
+
+    - name: Install Helm
+      uses: azure/setup-helm@v3
+      id: install
+      
     - name: Create GKE cluster
       run: |-
         repoName='kubernetes-clients-compatibility'
         clusterName="$repoName-$GITHUB_RUN_NUMBER"
         echo "clusterName=$clusterName" >> $GITHUB_ENV
-        gcloud container clusters create "$clusterName" --zone="$GKE_ZONE" --project="$PROJECT_ID" --machine-type=n1-standard-2 --num-nodes=2 --no-enable-ip-alias
+        gcloud container clusters create "$clusterName" \
+        --zone="$GKE_ZONE" \
+        --project=${{ secrets.PROJECT_ID }} \
+        --machine-type=n1-standard-2 \
+        --num-nodes=2 \
+        --network=${{env.GCP_NETWORK}} 
         sleep 30
+
     - name: Deploy Hazelcast cluster
       run: |-
-        kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast-kubernetes/master/rbac.yaml
-        kubectl create service loadbalancer hz-hazelcast-0 --tcp=5701
-        kubectl run hz-hazelcast-0 --image=hazelcast/hazelcast:$HAZELCAST_VERSION --port=5701 -l "app=hz-hazelcast-0,role=hazelcast"
-        kubectl create service loadbalancer hz-hazelcast-1 --tcp=5701
-        kubectl run hz-hazelcast-1 --image=hazelcast/hazelcast:$HAZELCAST_VERSION --port=5701 -l "app=hz-hazelcast-1,role=hazelcast"
-        kubectl create service loadbalancer hz-hazelcast-2 --tcp=5701
-        kubectl run hz-hazelcast-2 --image=hazelcast/hazelcast:$HAZELCAST_VERSION --port=5701 -l "app=hz-hazelcast-2,role=hazelcast"
-        kubectl create service loadbalancer hz-hazelcast --tcp=5701 -o yaml --dry-run=client | kubectl set selector --local -f - "role=hazelcast" -o yaml | kubectl create -f -
+        helm repo add hazelcast https://hazelcast-charts.s3.amazonaws.com/
+        helm repo update
+        helm install operator hazelcast/hazelcast-platform-operator --set installCRDs=true       
+        cat <<EOF | kubectl apply -f -
+        apiVersion: hazelcast.com/v1alpha1
+        kind: Hazelcast
+        metadata:
+          name: hz-hazelcast
+        spec:
+          clusterSize: 3
+          exposeExternally:
+            type: Smart
+            discoveryServiceType: LoadBalancer
+            memberAccess: LoadBalancer
+        EOF
+        sleep 60
+
     - name: Wait for deployment to finish
-      run: |-
+      run: |-                    
+        echo "Waiting for the pods..."
+        while ! kubectl get pods hz-hazelcast-0; do sleep 2; done
+        echo "Pods are created."
         kubectl wait --for=condition=ready pod/hz-hazelcast-0 --timeout=180s
         kubectl wait --for=condition=ready pod/hz-hazelcast-1 --timeout=180s
         kubectl wait --for=condition=ready pod/hz-hazelcast-2 --timeout=180s
```

Comment:
I think you can just Hazelcast CR status to be sure about cluster state:
```
kubectl get hazelcast hz-hazelcast
```

Reply:
Fixed at https://github.com/hazelcast/client-compatibility-suites/pull/63/commits/3276c851e67e3c041e8181117efe9e646e1dffc5

Reply:
Fixed at https://github.com/hazelcast/client-compatibility-suites/pull/63/commits/4da235b1ad08992ebcbf184d077a20df7ee13eda